OpenRA.Mods.Common/Warheads/LeaveSmudgeWarhead.cs

```diff
@@ -38,7 +38,7 @@ public override void DoImpact(Target target, Actor firedBy, IEnumerable<int> dam
 			// Draw the smudges:
 			foreach (var sc in allCells)
 			{
-				var smudgeType = world.Map.GetTerrainInfo(sc).AcceptsSmudgeType.FirstOrDefault(t => SmudgeType.Contains(t));
+				var smudgeType = world.Map.GetTerrainInfo(sc).AcceptsSmudgeType.Intersect(SmudgeType).FirstOrDefault();
```

Comment:
This first is better as Intersects with create a new set internally for elements in the source collection. You could lose the lambda though ```FirstOrDefault(SmudgeType.Contains)```.